### ISSUE TITLE
Multi-arch docker images

### ISSUE THREAD

**Issue by vaibhavsood** [tensorflow/tensorflow#14934]
Are there non-x86 (ppc64le, arm etc) docker images available for tensorflow? 

This looks like an intel image : https://hub.docker.com/r/tensorflow/tensorflow/

**Comment by aselle** [comment]
Not to my knowledge @caisq.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 178 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 193 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 208 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 223 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 238 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by angersson** [comment]
I've recently proposed a change to TensorFlow that obsoletes parameterized_docker_build.sh, which may help alleviate this issue (because it could lead to more easily-maintained non-x86 images). If anyone following this thread is interested in making TensorFlow's Dockerfile story better for everyone, [please take a look at the RFC](https://github.com/tensorflow/community/pull/8).

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jayfurmanek** [comment]
Just a quick note here that @angersson has a PR up for the new docker build method.
https://github.com/tensorflow/tensorflow/pull/21291

Once this is merged we'll be working on using the current community environment for ppc64le to also build docker images.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by harshini-gadige** [comment]
Is this issue resolved ?

**Comment by jayfurmanek** [comment]
It is not. I plan on getting back to this shortly for the ppc64le image, which should lay a blueprint for other arch's.

**Comment by tjakob** [comment]
I'm starting to look at the ppc64le image

**Comment by tensorflowbutler** [comment]
Nagging Assignee @harshini-gadige: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tjakob** [comment]
I have a pull request that adds ppc64le docker files. 
https://github.com/tensorflow/tensorflow/pull/23194

**Comment by harshini-gadige** [comment]
Thank you @tjakob

**Comment by tensorflowbutler** [comment]
Nagging Assignee @harshini-gadige: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @harshini-gadige: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tjakob** [comment]
There was a new PR to rework the new docker build method some. https://github.com/tensorflow/tensorflow/pull/24051. I'm reworking my PR to use this newer method.

**Comment by harshini-gadige** [comment]
@tjakob  Please keep us posted about your new PR

**Comment by tjakob** [comment]
My new PR for the latest rework of the dockerfiles can be found here: https://github.com/tensorflow/tensorflow/pull/24180

**Comment by tjakob** [comment]
https://github.com/tensorflow/tensorflow/pull/24180 just merged. This should provide a place to start for other non-x86 docker images.

**Comment by jayfurmanek** [comment]
Thanks @tjakob  and @angersson !

**Comment by jayfurmanek** [comment]
By the way, the ppc64le docker images are published to:
https://hub.docker.com/r/ibmcom/tensorflow-ppc64le/

**Comment by Revanee** [comment]
Any updates on this? Currently all of the official containers are linux/amd64. Are there any plans to support multiple architectures?

**Comment by rposts** [comment]
@tjakob and @angerson - I would like to investigate the possibility of adding s390x arch images (be) in the mix.  Just wondering if the TF CI infrastructure now supports multi-arch Docker images?  I could not spot `ppc64_le` Docker images on tensorflow/Dockerhub repo.  Are there plans of publishing multi-arch images on tensorflow repo on Dockerhub? Thanks.

**Comment by angerson** [comment]
> Are there plans of publishing multi-arch images on tensorflow repo on Dockerhub? Thanks.

We don't have plans for this. Unfortunately, due to loss of staff on our team, it's more likely that we'd have to shrink our Docker support instead, but there are no plans as of now.

**Comment by kun-lu20** [comment]
Hi @angerson and @tjakob ,

Do you have any updates reg plans of publishing multi-arch images on tensorflow repo? Thanks

**Comment by angerson** [comment]
We still have no plans for multi-arch Docker images.

**Comment by SuryanarayanaY** [comment]
Hi @vaibhavsood,

I think this request for Multi-arch docker images may not be considered due to resource constraints. Do you still feel this ticket needs to be in open condition ? Iam adding contributions welcome label again to make it visible for community.

Thanks!

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow version with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the release notes to stay up to date with the latest developments which are happening in the Tensorflow space.

Thank you!
